internal/agent/device/applications/sim_manager.go

```diff
@@ -0,0 +1,68 @@
+package applications
+
+import (
+	"context"
+
+	"github.com/flightctl/flightctl/api/v1alpha1"
+	"github.com/flightctl/flightctl/internal/agent/device/applications/provider"
+	"github.com/flightctl/flightctl/internal/agent/device/dependency"
+	"github.com/flightctl/flightctl/internal/agent/device/status"
+	"github.com/flightctl/flightctl/pkg/log"
+	"github.com/samber/lo"
+)
+
+var _ Manager = (*simManager)(nil)
+
+type simManager struct {
```

Comment:
the agent should not care or reason about simulation. please reconsider this approach. we have mocks if we wanted to have the managers do no-ops. but then I wonder why use the agent at all.

Reply:
I am OK of delaying it for now.

Reply:
Ok, I've rolled back the changes.